src/mpsc/blocking.rs

```diff
@@ -351,6 +351,124 @@ feature! {
+        /// reused in place.
+        ///
+        /// # Errors
+        ///
+        /// - [`Err`]`(`[`SendTimeoutError::Timeout`]`)` if the timeout has elapsed.
+        /// - [`Err`]`(`[`SendTimeoutError::Closed`]`)` if the channel has closed.
+        ///
+        /// # Examples
+        ///
+        /// Sending formatted strings by writing them directly to channel slots,
+        /// in place:
+        ///
+        /// ```
+        /// use thingbuf::mpsc::{blocking::StaticChannel, errors::SendTimeoutError};
+        /// use std::{fmt::Write, time::Duration, thread};
+        ///
+        /// static CHANNEL: StaticChannel<String, 1> = StaticChannel::new();
+        /// let (tx, rx) = CHANNEL.split();
+        ///
+        /// thread::spawn(move || {
+        ///     thread::sleep(Duration::from_millis(500));
+        ///     let msg = rx.recv_ref().unwrap();
+        ///     println!("{}", msg);
+        ///     thread::sleep(Duration::from_millis(500));
+        /// });
+        ///
+        /// thread::spawn(move || {
+        ///     let mut value = tx.send_ref_timeout(Duration::from_millis(200)).unwrap();
+        ///     write!(value, "hello").expect("writing to a `String` should never fail");
+        ///     thread::sleep(Duration::from_millis(400));
+        ///
+        ///     let mut value = tx.send_ref_timeout(Duration::from_millis(200)).unwrap();
+        ///     write!(value, "world").expect("writing to a `String` should never fail");
+        ///     thread::sleep(Duration::from_millis(400));
+        ///
+        ///     assert_eq!(
+        ///         Err(&SendTimeoutError::Timeout(())),
+        ///         tx.send_ref_timeout(Duration::from_millis(200)).as_deref().map(String::as_str)
+        ///     );
+        /// });
+        /// ```
+        ///
+        /// [`send`]: Self::send
+        #[cfg(not(all(test, loom)))]
+        pub fn send_ref_timeout(&self, timeout: Duration) -> Result<SendRef<'_, T>, SendTimeoutError> {
+            send_ref_timeout(self.core, self.slots, self.recycle, timeout)
+        }
+
+        /// Sends a message by value, blocking until there is a free slot to
+        /// write to, waiting for at most `timeout`.
+        ///
+        /// This method takes the message by value, and replaces any previous
+        /// value in the slot. This means that the channel will *not* function
+        /// as an object pool while sending messages with `send`. This method is
+        /// most appropriate when messages don't own reusable heap allocations,
+        /// or when the [`StaticReceiver`] end of the channel must receive messages
+        /// by moving them out of the channel by value (using the
+        /// [`StaticReceiver::recv`] method). When messages in the channel own
+        /// reusable heap allocations (such as `String`s or `Vec`s), and the
+        /// [`StaticReceiver`] doesn't need to receive them by value, consider using
+        /// [`send_ref`] instead, to enable allocation reuse.
```

Comment:
let's reference `send_ref_timeout` here, since it's the equivalent that has a timeout?
```suggestion
        /// [`send_ref_timeout`] instead, to enable allocation reuse.
```